Content.Server/Power/EntitySystems/ChargerSystem.cs

```diff
@@ -13,69 +14,69 @@ internal sealed class ChargerSystem : EntitySystem
 {
     [Dependency] private readonly ItemSlotsSystem _itemSlotsSystem = default!;
     [Dependency] private readonly PowerCellSystem _cellSystem = default!;
+    [Dependency] private readonly SharedAppearanceSystem _sharedAppearanceSystem = default!;
 
     public override void Initialize()
     {
-        SubscribeLocalEvent<ChargerComponent, ComponentInit>(OnChargerInit);
-        SubscribeLocalEvent<ChargerComponent, ComponentRemove>(OnChargerRemove);
+        SubscribeLocalEvent<ChargerComponent, ComponentStartup>(OnStartup);
         SubscribeLocalEvent<ChargerComponent, PowerChangedEvent>(OnPowerChanged);
         SubscribeLocalEvent<ChargerComponent, EntInsertedIntoContainerMessage>(OnInserted);
         SubscribeLocalEvent<ChargerComponent, EntRemovedFromContainerMessage>(OnRemoved);
         SubscribeLocalEvent<ChargerComponent, ContainerIsInsertingAttemptEvent>(OnInsertAttempt);
         SubscribeLocalEvent<ChargerComponent, ExaminedEvent>(OnChargerExamine);
     }
 
+    private void OnStartup(EntityUid uid, ChargerComponent component, ComponentStartup args)
+    {
+        UpdateStatus(uid, component);
+    }
+
     private void OnChargerExamine(EntityUid uid, ChargerComponent component, ExaminedEvent args)
     {
         args.PushMarkup(Loc.GetString("charger-examine", ("color", "yellow"), ("chargeRate", component.ChargeRate)));
     }
 
     public override void Update(float frameTime)
     {
-        foreach (var comp in EntityManager.EntityQuery<ChargerComponent>())
+        foreach (var comp in EntityManager.EntityQuery<ChargerActiveComponent>())
```

Comment:
```suggestion
        foreach (var (_, charger, slotComp) in EntityManager.EntityQuery<ChargerActiveComponent, ChargerComponent, ItemSlotsComponent>())
```

Also you have a return in the loop that should be a continue.